### PR TITLE
pass error correctly in index.dropCloudShards

### PR DESCRIPTION
**Pull request by moogacs**
### What's being changed:
follow up for https://github.com/weaviate/weaviate/pull/5288 

### Review checklist

- [ ] Documentation has been updated, if necessary. Link to changed documentation:
- [ ] Chaos pipeline run or not necessary. Link to pipeline:
- [ ] All new code is covered by tests where it is reasonable.
- [ ] Performance tests have been run or not necessary.

### CODE REVIEW

**Review by antas-marcin — DISMISSED** [review]

File: adapters/repos/db/index.go [hunk 1]
```diff
@@ -2002,7 +2002,13 @@ func (i *Index) dropCloudShards(ctx context.Context, cloud modulecapabilities.Of
...
 			defer i.shardCreateLocks.Unlock(name)
 
-			return cloud.Delete(ctx, i.ID(), name, nodeId)
+			err := cloud.Delete(ctx, i.ID(), name, nodeId)
+			if err != nil {
+				ec.Add(err)
+				i.logger.WithField("action", "cloud_drop_shard").
+					WithField("shard", name).Error(err)
+			}
+			return err
```
Comment (antas-marcin) [inline adapters/repos/db/index.go:2011]: you are adding this error to `ec.Add(err)` but returning `err` is that ok?

Comment (antas-marcin) [reply adapters/repos/db/index.go:2011]: shouldn't be there `return ec.ToError()`?

Comment (moogacs) [reply adapters/repos/db/index.go:2011]: yeah return the error os for eg and ec for the whole function

Comment (moogacs) [reply adapters/repos/db/index.go:2011]: actually it should be nil, so that eg will continue

**Review by antas-marcin — APPROVED** [review]

**Pull request merged**